src/Form/Form.php

```diff
@@ -209,11 +210,32 @@ protected function _buildSchema(Schema $schema): Schema
      * Used to check if $data passes this form's validation.
      *
      * @param array $data The data to check.
+     * @param array $options The validate options
      * @return bool Whether or not the data is valid.
+     * @throws \RuntimeException If validator is invalid.
      */
-    public function validate(array $data): bool
+    public function validate(array $data, array $options = []): bool
     {
-        $validator = $this->getValidator();
+        $options += [
+            'validate' => true,
+        ];
+
+        $validator = null;
+        if ($options['validate'] === true) {
+            $validator = $this->getValidator();
+        } elseif (is_string($options['validate'])) {
+            $validator = $this->getValidator($options['validate']);
+        } elseif (is_object($options['validate'])) {
```

Comment:
I know that you have copied this code block verbatim from `Marshaller::_validate()` but I don't see the need to support passing a validator object. (Even for the marshaller/table it's not well documented and should be just deprecated/removed but that's a separate issue.)

With the validator name only thing configurable the instead of `$options` name we could just have `?string $validator` as the argument. Then the only change required will be `$validator = $this->getValidator($validator);` instead of `$validator = $this->getValidator();`

Reply:
I'm okay with that. Waiting others comments and i will clean code, or close PR ;)

Reply:
I think having the `validate` option is ok. It is not something that will get used often but it also doesn't add 'new' syntax as we have a similar option and usage in ORM tables.

I agree with @ADmad that we don't need to support passing a validator object in, and that we should probably deprecate that behavior from Table as well.